marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/graph/searchNodes.sjs

```diff
@@ -162,13 +162,12 @@ result.map(item => {
     //Override if count is more than 1. We will have a node with badge.
     if (item.nodeCount > 1) {
       let entityType = objectIRIArr[objectIRIArr.length - 2];
-      objectIRI = entityType;
       objectLabel =  entityType;
       objectUri = null;
       objectId = item.subjectIRI.toString() + "-" + objectIRIArr[objectIRIArr.length - 2];
     }
     let edge = {};
-    edge.id = "edge-" + item.subjectIRI + "-" + item.predicateIRI + "-" + objectIRI;
+    edge.id = "edge-" + item.subjectIRI + "-" + item.predicateIRI + "-" + objectId;
```

Comment:
Careful here.
If objectId = item.subjectIRI.toString() + "-" + objectIRIArr[objectIRIArr.length - 2];
then the edge id would have duplicated text and be really long.

Reply:
Think that change is my fault. I noticed that the grouped nodes weren't connecting by testing manually and copying/pasting the nodes and edges into a VisJs JSfiddle.  In order to make it work, I had to change the endpoint to use objectId a couple of places. I may have been overzealous in making that change. I'll adjust this.

Reply:
Fixed